README.md

```diff
@@ -1,24 +1,158 @@
-Serverless IAM via Terraform
-============================
+AWS Serverless Module
+=====================
+[![Terraform][tf_img]][tf_site]
+[![Travis Status][trav_img]][trav_site]
 
-Tuned, locked-down IAM roles for the minimum set of privileges to run the `serverless` framework.
+Get your [serverless][] framework application to AWS, the **right way**.
 
+## Contents
 
-## Installation
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 
-First, [download Terraform](https://www.terraform.io/downloads.html) and [install the binary](https://www.terraform.io/intro/getting-started/install.html). You should be able to run:
 
-```sh
-$ terraform --version
+- [Overview](#overview)
+- [Concepts](#concepts)
+- [Modules](#modules)
+- [Integration](#integration)
+  - [Reference project](#reference-project)
+  - [Module integration](#module-integration)
+  - [AWS IAM group integration](#aws-iam-group-integration)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+Getting a [serverless][] application all the way to production in AWS **correctly** using **securely** can be quite challenging. In particular, things like:
+
+- Locking down IAM permissions to the minimum needed for different conceptual "roles" (e.g., `admin`, `developer`, `ci`).
+- Providing a scheme for different environments/stages (e.g., `development`, `staging`, `production`).
+
+... lack reasonable guidance to practically achieve in real world applications.
+
+This [Terraform][] module provides a production-ready base of AWS permissions / resources to support a `serverless` framework application and help manage development / deployment workflows and maintenance. Specifically, it provides:
+
+- **IAM Groups**: Role-specific groups to attach to AWS users to give humans and CI the minimum level of permissions locked to both a specific `serverless` service and stage/environment.
+
+## Concepts
+
+This module allows practical isolation / compartmentalization of privileges within a single AWS account along the folowing axes:
+
+* **Stage/Environment**: An arbitrary environment to isolate -- this module doesn't restrict selection in any way other than there has to be at least one. In practice, a good set of choices may be something like `sandbox`, `development`, `staging`, `production`.
+* **IAM Groups**: This module creates/enforces a scheme wherein:
+    * **Admin**: AWS users assigned to the `admin` group can create/update/delete a `serverless` application and do pretty much anything that the `serverless` framework permits out of the box.
+    * **Developer, CI**: AWS users assigned to the `developer|ci` groups can update a `serverless` application and do non-mutating things like view logs, perform rollbacks, etc.
```

Comment:
Nit: `read-only` might be more clear than `non-mutating`

Reply:
For this one, `rollback` is more than read-only. And it actually somewhat mutates too...

Maybe just replace with `do things like`...?

Reply:
Oh! Yeah that works!